src/lua/uri.lua

```diff
@@ -226,7 +226,11 @@ local function format(uri, write_password)
     uribuf.scheme = uri.scheme
     uribuf.login = uri.login
     uribuf.password = uri.password
-    uribuf.host = uri.host
+    if uri.ipv6 ~= nil then
+        uribuf.host = string.format("[%s]", uri.host)
+    else
+        uribuf.host = uri.host
+    end
```

Comment:
Let's split parsing and formatting fixes into own commits.

Reply:
corrected